### PR TITLE
[SPARK-41400][CONNECT] Remove Connect Client Catalyst Dependency

### PR DESCRIPTION
**Pull request by hvanhovell**
### What changes were proposed in this pull request?
This PR decouples the Spark Connect Scala Client from Catalyst, it now used SQL API module instead.

There were quite a few changes we still needed to make:
- For testing we needed a bunch of utilities. I have moved these to common-utils.
- I have moved bits and pieces of IntervalUtils to SparkIntervalUtils.
- A lot of small fixes.

### Why are the changes needed?
This reduces the client's dependency tree from ~300 MB of deps to ~30MB. This makes it easier to use the client when you are developing connect applications. On top of this the reduced dependency graph also means folks will be less affected by the clients' classpath.

### Does this PR introduce _any_ user-facing change?
Yes. It changes the classpath exposed by the Spark Connect Scala Client.

### How was this patch tested?
Existing tests.

### CODE REVIEW

**Comment by hvanhovell** [comment]
This need https://github.com/apache/spark/pull/42164 to go in first.

File: common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java [hunk 1]
```diff
@@ -416,7 +416,7 @@ public void initializeApplication(ApplicationInitializationContext context) {
     String appId = context.getApplicationId().toString();
     try {
       ByteBuffer appServiceData = context.getApplicationDataForService();
-      String payload = JavaUtils.bytesToString(appServiceData);
+      String payload = JavaUtils.bytesToString(appServiceData.slice());
```
Comment (hvanhovell) [inline common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java:419]: This was an actual bug, that it used to work was luck - the . The buffer is consumed twice, here and later on when we serialize the password. In that case the second serialization will read an exhausted buffer.

**Comment by hvanhovell** [comment]
cc @amaliujia @cloud-fan @zhenlineo

**Comment by amaliujia** [comment]
It is happening!

**Review by amaliujia — APPROVED** [review]

**Comment by hvanhovell** [comment]
Merging this. Single test failure, is a bit flaky.

**Pull request closed without merging**